consul/src/main/scala/io/buoyant/consul/v1/ConsulApi.scala

```diff
@@ -0,0 +1,158 @@
+
+  def serviceNodes(
+    serviceName: String,
+    datacenter: Option[String] = None,
+    tag: Option[String] = None,
+    blockingIndex: Option[String] = None,
+    retry: Boolean = false
+  ): Future[Indexed[Seq[ServiceNode]]]
+}
+
+object CatalogApi {
+  def apply(c: Client): CatalogApi = new CatalogApi(c, s"/$versionString")
+}
+
+class CatalogApi(
+  val client: Client,
+  val uriPrefix: String,
+  val backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+  val stats: StatsReceiver = DefaultStatsReceiver
+) extends ConsulApi with Closable {
+
+  val catalogPrefix = s"$uriPrefix/catalog"
+
+  // https://www.consul.io/docs/agent/http/catalog.html#catalog_datacenters
+  def datacenters(retry: Boolean = false): Future[Seq[String]] = {
+    val req = mkreq(http.Method.Get, s"$catalogPrefix/datacenters")
+    executeJson[Seq[String]](req, retry).map(_.value)
+  }
+
+  // https://www.consul.io/docs/agent/http/catalog.html#catalog_services
+  def serviceMap(
+    datacenter: Option[String] = None,
+    blockingIndex: Option[String] = None,
+    retry: Boolean = false
+  ): Future[Indexed[Map[String, Seq[String]]]] = {
+    val req = mkreq(
+      http.Method.Get,
+      s"$catalogPrefix/services",
+      "index" -> blockingIndex,
+      "dc" -> datacenter
+    )
+    executeJson[Map[String, Seq[String]]](req, retry)
+  }
+
+  // https://www.consul.io/docs/agent/http/catalog.html#catalog_service
+  def serviceNodes(
+    serviceName: String,
+    datacenter: Option[String] = None,
+    tag: Option[String] = None,
+    blockingIndex: Option[String] = None,
+    retry: Boolean = false
+  ): Future[Indexed[Seq[ServiceNode]]] = {
+    val req = mkreq(
+      http.Method.Get,
+      s"$catalogPrefix/service/$serviceName",
+      "index" -> blockingIndex,
+      "dc" -> datacenter,
+      "tag" -> tag
+    )
+    executeJson[Seq[ServiceNode]](req, retry)
+  }
+}
+
+object HealthApi {
+  def apply(c: Client): HealthApi = new HealthApi(c, s"/$versionString")
+}
+
+class HealthApi(
+  override val client: Client,
+  override val uriPrefix: String,
+  override val backoffs: Stream[Duration] = Backoff.exponentialJittered(1.milliseconds, 5.seconds),
+  override val stats: StatsReceiver = DefaultStatsReceiver
+) extends CatalogApi(
+  client,
+  uriPrefix,
+  backoffs,
+  stats
+) with Closable {
+
+  val healthPrefix = s"$uriPrefix/health"
+
+  // https://www.consul.io/docs/agent/http/health.html#health_service
+  override def serviceNodes(
+    serviceName: String,
+    datacenter: Option[String] = None,
+    tag: Option[String] = None,
+    blockingIndex: Option[String] = None,
+    retry: Boolean = false
+  ): Future[Indexed[Seq[ServiceNode]]] = {
+    val req = mkreq(
+      http.Method.Get,
+      s"$healthPrefix/service/$serviceName",
+      "index" -> blockingIndex,
+      "dc" -> datacenter,
+      "tag" -> tag,
+      "passing" -> Some("true")
+    )
+    executeJson[Seq[ServiceHealth]](req, retry).map(
```

Comment:
I have an unexpected response exception when I run locally.  I'm pretty sure it's the same reason you've pointed out.  I'll send a follow up when I get a chance to dig in.